### PR TITLE
moving IEstimator, ITransformer ,IDataReader,  IDataReaderEstimator, …

### PR DESCRIPTION
**Pull request by sfilipi**
moving IEstimator, ITransformer, IDataReader,  IDataReaderEstimator, SchemaShape from Microsoft.ML.Core.Data to Microsoft.ML. 
Those were the only types in Microsoft.ML.Core.Data. 

The change is mechanical, IMO you can just approve if the above sentence is not controversial. 

Towards #2326.

### CODE REVIEW

File: src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs [hunk 1]
```diff
@@ -27,7 +26,7 @@
 [assembly: LoadableClass(typeof(void), typeof(KMeansPlusPlusTrainer), null, typeof(SignatureEntryPointModule), "KMeans")]
 
 namespace Microsoft.ML.Trainers.KMeans
-{
+    {
```
Comment (rogancarr) [inline src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs]: Wrong indent?

File: src/Microsoft.ML.PCA/PcaTrainer.cs [hunk 2]
```diff
@@ -31,7 +30,7 @@
 [assembly: LoadableClass(typeof(void), typeof(RandomizedPcaTrainer), null, typeof(SignatureEntryPointModule), RandomizedPcaTrainer.LoadNameValue)]
 
 namespace Microsoft.ML.Trainers.PCA
-{
+    {
```
Comment (rogancarr) [inline src/Microsoft.ML.PCA/PcaTrainer.cs]: Wrong indent?

File: src/Microsoft.ML.PCA/PcaTransformer.cs [hunk 3]
```diff
@@ -32,7 +31,7 @@
 [assembly: LoadableClass(typeof(void), typeof(PrincipalComponentAnalysisTransformer), null, typeof(SignatureEntryPointModule), PrincipalComponentAnalysisTransformer.LoaderSignature)]
 
 namespace Microsoft.ML.Transforms.Projections
-{
+    {
```
Comment (rogancarr) [inline src/Microsoft.ML.PCA/PcaTransformer.cs]: Wrong indent?

File: test/Microsoft.ML.Tests/TrainerEstimators/LbfgsTests.cs [hunk 4]
```diff
@@ -3,14 +3,13 @@
...
 using Microsoft.Data.DataView;
-using Microsoft.ML.Core.Data;
 using Microsoft.ML.Data;
 using Microsoft.ML.Internal.Calibration;
 using Microsoft.ML.Trainers;
 using Xunit;
 
 namespace Microsoft.ML.Tests.TrainerEstimators
-{
+    {
```
Comment (rogancarr) [inline test/Microsoft.ML.Tests/TrainerEstimators/LbfgsTests.cs]: Bad indent?

File: test/Microsoft.ML.Tests/TrainerEstimators/TrainerEstimators.cs [hunk 5]
```diff
@@ -15,7 +13,7 @@
 using Xunit.Abstractions;
 
 namespace Microsoft.ML.Tests.TrainerEstimators
-{
+    {
```
Comment (rogancarr) [inline test/Microsoft.ML.Tests/TrainerEstimators/TrainerEstimators.cs]: Bad indent?

**Comment by sfilipi** [comment]
@rogancarr I just noticed the indentation notes. I'll address them on the next pr...

**Pull request merged**